code/controller/solenoid.cpp

```diff
@@ -141,6 +143,46 @@ const QSerialPort &Solenoid::serial_port() const {
     return m_serial;
 }
 
+void Solenoid::change_up_power(int value) {
+    log() << "Changing up solenoid power to " << value;
```

Comment:
there's some repeated code here

you can add a generic function `change_power(int value, Controller::Dir dir)`

Reply:
I wasn't sure how, since I am using these as slots which are called when the slider value is changed, they are only passed the value and I'm not sure how to get the direction.